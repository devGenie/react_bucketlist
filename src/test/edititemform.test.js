import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount } from 'enzyme';
import {stub, spy} from 'sinon';
import sinon from 'sinon';
import 'should';
import 'should-enzyme';
import toJson from 'enzyme-to-json';
import sinonStubPromise from 'sinon-stub-promise';

import EditItem from '../components/edit-item';
import DashBoard from '../components/dashboard';
import localStorageMock from '../__mocks__/localStorage';

sinonStubPromise(sinon)
window.sessionStorage = localStorageMock;

describe('<BucketlistForm/>',()=>{
	let formMount;
	let fetchMock;
	let changeSpy;
	let submitSpy;
	let caller;

	beforeEach(()=>{
		fetchMock = stub(window,'fetch').returnsPromise().resolves({status:'success'});
		caller = spy(DashBoard.prototype,'handleFetchCaller');
		changeSpy = spy(EditItem.prototype,'handleChange');
		submitSpy = spy(EditItem.prototype,'handleSubmit');
		formMount = mount(<EditItem caller={{bucketlist:'1',item:{id:'1',name:'one'},callback:''}}/>);
	});

	afterEach(()=>{
		fetchMock.restore();
		changeSpy.restore();
		submitSpy.restore();
		caller.restore();
	});

	it('can render correctly',()=>{
		expect(toJson(formMount)).toMatchSnapshot()
	});

	it('can submit data',()=>{
		formMount.find('form').simulate('submit',{
			preventDefault:()=>{}
		});
		expect(submitSpy.called).toBe(true)
	});

	it('can handle change event',()=>{
		formMount.find('form').find({name:'name'}).simulate('change',{
			target:{
				value:'changed'
			}
		});
		expect(changeSpy.called).toBe(true)
	})

	
})