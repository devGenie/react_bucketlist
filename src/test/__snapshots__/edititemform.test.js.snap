// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BucketlistForm/> can render correctly 1`] = `
<EditItem
  caller={
    Object {
      "bucketlist": "1",
      "callback": "",
      "item": Object {
        "id": "1",
        "name": "one",
      },
    }
  }
>
  <div
    className="modal"
    id="item_edit_model"
  >
    <form
      className="modal-content"
      method="POST"
      onSubmit={[Function]}
    >
      <h4
        className="header2"
      >
        Edit Item
      </h4>
      <FieldWrapper
        Label="Name"
      >
        <div
          className="input-field  "
        >
          <input
            className="validate"
            id="item_name"
            name="name"
            onChange={[Function]}
            placeholder="Name of bucketlist item"
            type="text"
            value="one"
          />
          <label>
            Name
          </label>
        </div>
      </FieldWrapper>
      <FieldWrapper
        extraz="right"
      >
        <div
          className="input-field right"
        >
          <button
            className="btn"
            type="submit"
          >
            Edit
          </button>
          <label />
        </div>
      </FieldWrapper>
    </form>
  </div>
</EditItem>
`;
